test/features/app/spec.js

```diff
@@ -29,9 +28,19 @@ function staticSpecLoad() {
     }
 
     return spec;
-
 }
 
+function initSpecAndParams() {
+    server = new Server();
+    return server.start()
```

Comment:
It's a bit more complicated Some tests are created programmatically and test creation runs first before before() is even executed. Since the test relies on the server config, the test creation fails. Spinning up an instance seemed like a simple compromise to get the settings and then execute the test creation. I could leave that instance running and then shut it at the end.

Reply:
So, [what we do](https://github.com/wikimedia/restbase/blob/master/test/features/specification/monitoring.js) in RESTBase for a similar set of tests is construct dynamic tests from within the first test case (should get the spec) which solves this problem. And it's ok since the tests are generated from the spec, so if the spec fetching test fails, we wouldn't be able to construct the tests anyway.

This test suite and RESTBase one are actually doing the same thing and it's a copy-pasta code. Maybe we should think about moving most of it into some library?

Reply:
I am going to leave this one open for now in case you have a better idea. I am still spinning up a server to load the spec.